chainer/functions/connection/n_step_rnn.py

```diff
@@ -161,8 +173,20 @@ def __init__(self, n_layers, states, rnn_dir, rnn_mode, **kwargs):
             candidate_list = ','.join(_rnn_modes.keys())
             raise ValueError('Invalid rnn_mode: "%s". Please select from [%s]'
                              % (rnn_mode, candidate_list))
+
+        is_enable_cudnn_v6 = isinstance(_rnn_persistent_algo, dict)
+        if is_enable_cudnn_v6 and rnn_algo not in _rnn_persistent_algo:
+            candidate_list = ','.join(_rnn_persistent_algo.keys())
+            raise ValueError('Invalid rnn_algo: "%s". Please select from [%s]'
+                             % (rnn_algo, candidate_list))
+            _rnn_algo = _rnn_persistent_algo[rnn_algo]
+        else:
+            _rnn_algo = None
```

Comment:
With this code, when a user uses older cuDNN, rnn_algo argument is ignore silently, isn't it? It is better to notify this argument is unavailable in such situation. Check if a user specify 'static' or 'dynamic' and when he/she uses older cuDNN raise an error.